src/Sylius/Bundle/PayumBundle/spec/Validator/GatewayFactoryExistsValidatorSpec.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\PayumBundle\Validator;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\PayumBundle\Model\GatewayConfigInterface;
+use Sylius\Bundle\PayumBundle\Validator\Constraints\GatewayFactoryExists;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Context\ExecutionContextInterface;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+use Symfony\Component\Validator\Violation\ConstraintViolationBuilderInterface;
+
+final class GatewayFactoryExistsValidatorSpec extends ObjectBehavior
```

Comment:
Could you add a test case for the `offline` factory name scenario?

Reply:
resolving as the offline case is no more valid